packages/x-data-grid/src/models/gridRows.ts

```diff
@@ -116,9 +116,9 @@ export interface GridDataGroupNode extends GridBasicGroupNode {
 
 export interface GridDataSourceGroupNode extends GridDataGroupNode {
   /**
-   * If true, this node has children on server.
+   * Number of children this node has on the server. Returns `-1` if unknown.
    */
-  hasServerChildren: boolean;
+  serverChildrenCount?: number;
```

Comment:
Shouldn't it always have a value for the `GridDataSourceGroupNode`, at least as per the [signature](https://github.com/mui/mui-x/blob/8534f2db66707596a1aad7e641826124fa16e6a0/packages/x-data-grid/src/models/gridDataSource.ts#L72-L77) of `getChildrenCount` prop?

Did you find some cases where it could be passed as `undefined`?
```suggestion
  serverChildrenCount: number;
```

Reply:
This was a mistake, fixed...